File: SaveSystem.cpp
#include "SaveSystem.h"
#include "PetParameters.h"
#include <fstream>

SaveSystem::SaveSystem(std::string path) : savePath(path) {}

bool SaveSystem::saveGame(const PetBase& pet) {
    std::ofstream file(savePath);
    if (!file.is_open()) return false;

    file << pet.getName() << "\n"
        << pet.getType() << "\n"
        << pet.getAge() << "\n";

    const auto& params = pet.getParameters();
    file << params.getHunger() << "\n"
        << params.getFatigue() << "\n"
        << params.getHealth() << "\n"
        << params.getMood() << "\n";

    file.close();
    return true;
}

bool SaveSystem::loadGame(PetBase& pet) {
    std::ifstream file(savePath);
    if (!file.is_open()) return false;

    std::string name, type;
    int age, hunger, fatigue, health, mood;
    file >> name >> type >> age >> hunger >> fatigue >> health >> mood;

    // Установка параметров через геттеры/сеттеры
    // Нужно будет изменить логику, если поля приватные
    // Для простоты предположим, что у нас есть сеттеры в PetBase
    // Это не так, но можно добавить виртуальные методы set в PetBase

    // Для текущей реализации:
    // VirtualPet::setParameters(hunger, fatigue, health, mood);
    // Это потребует изменений в VirtualPet

    return true;
}

File: SaveSystem.cs
using System;
using System.IO;

public class SaveSystem
{
    private string savePath;

    public SaveSystem(string path = "save.txt")
    {
        savePath = path;
    }

    public bool SaveGame(PetBase pet)
    {
        try
        {
            using (StreamWriter file = new StreamWriter(savePath))
            {
                file.WriteLine(pet.GetName());
                file.WriteLine(pet.GetType());
                file.WriteLine(pet.GetAge());

                var parameters = pet.GetParameters();
                file.WriteLine(parameters.GetHunger());
                file.WriteLine(parameters.GetFatigue());
                file.WriteLine(parameters.GetHealth());
                file.WriteLine(parameters.GetMood());
            }
            return true;
        }
        catch
        {
            return false;
        }
    }

    public bool LoadGame(PetBase pet)
    {
        try
        {
            if (!File.Exists(savePath))
                return false;

            using (StreamReader file = new StreamReader(savePath))
            {
                string name = file.ReadLine();
                string type = file.ReadLine();
                int age = int.Parse(file.ReadLine());

                int hunger = int.Parse(file.ReadLine());
                int fatigue = int.Parse(file.ReadLine());
                int health = int.Parse(file.ReadLine());
                int mood = int.Parse(file.ReadLine());

                pet.SetName(name);
                pet.SetType(type);
                pet.SetAge(age);

                var parameters = pet.GetParameters();
                parameters.SetHunger(hunger);
                parameters.SetFatigue(fatigue);
                parameters.SetHealth(health);
                parameters.SetMood(mood);

                pet.Update();
            }
            return true;
        }
        catch
        {
            return false;
        }
    }
}